internal/outgoing/dispatcher.go

```diff
@@ -315,11 +315,21 @@ func (d *Dispatcher) processDelayedEvent(e models.IncomingEvent, timeRemain time
 		// Schedule send event later
 		dispatcher.TrackAfterFuncJob(timeRemain, func() {
 			dLogger.WithFields(log.Fields{"key": dataKey}).Debug("After event callback")
-			payload, _, _, _ := store.Get(dataKey)
+			payload, exists, _, _ := store.Get(dataKey)
+			// Key might be deleted by another worker, resulting in data not found
+			if !exists {
+				dLogger.WithFields(log.Fields{"key": dataKey}).Debug("Event data not found")
+				d.OnError(models.IncomingEvent{}, &captin_errors.UnretryableError{Msg: "Event data not found", Event: e, Destination: dest})
+				return
+			}
 			event := models.IncomingEvent{}
 			json.Unmarshal([]byte(payload), &event)
 			d.sendEvent(event, dest, store, documentStore)
-			store.Remove(dataKey)
+
+			/* This line is commented out because their is a possibility of race condition,
+			when the dataKey is removed by one worker, another worker might try to access it,
+			it is better to let the dataKey expire by the ttl */
+			// store.Remove(dataKey)
```

Comment:
I think it's fine to keep that, if the data key is being removed the existence check above will return false and being early returned

Reply:
fixed